include/tvm/runtime/container.h

```diff
@@ -847,6 +847,19 @@ class Array : public ObjectRef {
  public:
   // Array's own methods
 
+  /*!
+   * \brief Concat two Arrays.
+   * \param lhs first Array to be concatenated.
+   * \param lhs second Array to be concatenated.
+   * \return The concatenated Array. Original Arrays are kept unchanged.
+   */
+  static Array<T> Concat(Array<T> lhs, const Array<T>& rhs) {
```

Comment:
We can make it a global template fucntion, so you can do Concat(lhs, rhs) without having to refer to the original type. Note that the type signature should be able to only run the function for the specific type so we won't risk over generalization